src/components/LoadingIndicator/LoadingIndicator.test.tsx

```diff
@@ -1,48 +1,103 @@
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
+    const messageContainer = container.getElementsByClassName('zui-loading-indicator-message');
+
+    expect(loadingIndicatorContainer).not.toContain(messageContainer);
+    expect(messageContainer.length).toBe(0);
+  });
 });
 
-test('should render text with custom postiion', () => {
-  const { getByTestId } = renderComponent({ text: mockText, spinnerPosition: 'left' });
-  const text = getByTestId(TEST_ID.TEXT);
-  expect(text).toHaveClass('zui-loading-indicator-text-left');
+describe('class names', () => {
+  // defined
+  test('should apply className prop to container class', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} />);
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
+
+    expect(loadingIndicatorContainer[0]).toHaveClass(MOCK_DEFAULT_PROPS.className);
+  });
+
+  test('should apply spinnerPosition prop to container class', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} />);
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
+
+    expect(loadingIndicatorContainer[0]).toHaveClass(`zui-loading-indicator-${MOCK_DEFAULT_PROPS.spinnerPosition}`);
+  });
+
+  test('should apply spinnerPosition prop to message container class', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} />);
+    const messageContainer = container.getElementsByClassName('zui-loading-indicator-message');
+
+    expect(messageContainer[0]).toHaveClass(`zui-loading-indicator-message-${MOCK_DEFAULT_PROPS.spinnerPosition}`);
+  });
+
+  // undefined
+  test('should not apply className prop to container class', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} className={undefined} />);
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
+
+    expect(loadingIndicatorContainer[0]).not.toHaveClass(MOCK_DEFAULT_PROPS.className);
+  });
+
+  test('should apply default "bottom" spinnerPosition to container class', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} spinnerPosition={undefined} />);
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
+
+    expect(loadingIndicatorContainer[0]).toHaveClass(`zui-loading-indicator-${MockSpinnerPositionType.BOTTOM}`);
+  });
+
+  test('should apply default "bottom" spinnerPosition to message container class', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} spinnerPosition={undefined} />);
+    const messageContainer = container.getElementsByClassName('zui-loading-indicator-message');
+
+    expect(messageContainer[0]).toHaveClass(`zui-loading-indicator-message-${MockSpinnerPositionType.BOTTOM}`);
+  });
```

Comment:
Per previous comment you can drop these tests after you remove the positioning classes from child elements.

Reply:
Updated!